.hooks/commit-msg

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+first_line=$(head -n1 "$1")
+echo "commit message: $first_line"
+
+conv_commit_msg="^(fix|feat|dep|ci|doc|refactor|test)(\(.{1,50}\))?: (.{1,100})\$"
```

Comment:
We also have to allow for the `!` before the colon in case of breaking changes (and I removed a unused capturing group):

```suggestion
conv_commit_msg="^(fix|feat|dep|ci|doc|refactor|test)(\(.{1,50}\))?!?: .{1,100}$"
```

One thing we could consider is being stricter on scope and summary, I think 150 characters in sum is too much and it shouldn't be allowed to contain all possible characters. I would propose `\w` plus ` ` and `-` (same for the summary) for allows characters.

```suggestion
conv_commit_msg="^(fix|feat|dep|ci|doc|refactor|test)(\([- \w]{1,15}\))?!?: [- \w]{5,70}$"
```

Reply:
I agree to restrict usage of special characters; I will try to keep it the one you mentioned and add a few more such as `_` and `/`. I would prefer to follow some guidelines around size of optional `scope` and mandatory `subject`. I myself did not come across any such guideline. Need not be something official, but something that is widely used/accepted. Did you find or know anything in that regard?

Reply:
> We also have to allow for the `!` before the colon in case of breaking changes (and I removed a unused capturing group):

Yes, this is a good point, I can mandate `BREAKING CHANGE` to be added in the trailer if `!` is specified.

Reply:
When I started our guide I didn't put any constraints on the scope because I felt like it heavily depends on the context. For example in a bosh release you will probably use different scopes compared to a pure source repository.

The subject should be a concise description of the change and the full length of the title line should be at most 70 characters, otherwise GitHub will [start wrapping it](https://github.com/therobinkim/commit-message-length/commit/788dd604d5b19b0bf9f0d8c254d4579d3b6d7b21) which hurts readability. Since the subject is a very important part of the commit message we should try to keep everything else short (hence `feat` and not `feature` as a prefix, same goes for the scope).

Reply:
I would not want to limit the type of text in scope and subject to "word characters", `-` and ` `. A bunch of other characters (;?/}{[]()*) and many more are excluded. Emoji are excluded.

Main goal is to have the release note created correctly from the commits. This is ensured with the type and breaking change indicator. The rest should be copied verbatim.